### ISSUE TITLE
If I click "copy" on pipeline detail, I cannot then save the new pipeline

### ISSUE THREAD

**Issue by tomas-knap** [mff-uk/odcs#965]

**Comment by bogo777** [comment]
Copy pipeline function from PipelineFacade (originaly used in PipelineList) already saves the pipeline during copy creation. So the copied pipeline is already saved and save buttons are disabled, since there are no changes since load of the copied pipeline. Is it ok? Or do we need to create copied pipeline without saving it to db?

**Comment by tomas-knap** [comment]
No, that is fine, it is just a bit confusing that it open a new pipeline record for me and it is automatically saved, which does not happen if I create new pipeline. 

Maybe we can change the name of the button: 
copy -> copy & open 

When clicking on such button, before copying and opening, we should inform user that the pipeline will be saved before it is copied. Obviously it may be saved only if the user has rights to it (it is not in read only mode).

And instead of having "copy & close" button, we can have only "copy", which creates a copy, notifies user about that, but user may remain working with the original pipeline. So it does not close the pipeline detail. 

What do you think?

**Comment by bogo777** [comment]
I think that there are 2 main motivations for copy button in pipeline edit:
1) Another user saved the same pipeline while I was editing it and I don't want to lose my changes but also don' t want to overwrite his. So I save my modifications as copy of the original pipeline.
2) I have only RO access to pipeline, so I must create my private copy before I can edit it.

For both these use cases, it is convenient how it works now. Maybe we could change the buttons caption if it seems confusing.

**Comment by tomas-knap** [comment]
Yes, understand the motivation. But it is also useful in situations when I open existing pipeline, do some changes and than I would like to store them to a new copy, rather then modifying the existing. 

Suggestion: 
-instead of "copy & close" button, we may have button with title "save a copy". Which only saves the copy, pipeline detail still shows the original pipeline. User may close by clicking on "close" button, so "& close" is not needed I think. 
-And button "copy" is actually "save a copy & open", but this title is rather long. Any suggestion?

**Comment by tomas-knap** [comment]
Bohuslav, please discuss

**Comment by bogo777** [comment]
Yes, understand the motivation. But it is also useful in situations when I open existing pipeline, do some changes and than I would like to store them to a new copy, rather then modifying the existing.

--Ok, but it works for situations like this, too. We save changes to new copy only. Maybe it is unclear that we open the copy also, but I think it is useful. If original pipeline is still open, user is left with bunch of unsaved changes, which he didn't want to save before(so he created copy instead) and what can he do with it? 

But if we open the copy, user can continue with editing it.

So maybe just make it more clear that copy is created & opened in the same step?

**Comment by tomas-knap** [comment]
We will finalize the discusion on the meeting next week
